dev/forbidden_from_release_tests/bin/main.dart

```diff
@@ -0,0 +1,272 @@
+    required this.dart,
+    required this.genSnapshot,
+    required this.feSnapshot,
+    required this.target,
+    required this.sdkProduct,
+    required this.packageConfig,
+    required this.packageName,
+    required this.forbiddenTypes,
+  });
+
+  factory Options.fromArgs(List<String> args) {
+    final String exe = Platform.isWindows ? '.exe' : '';
+    final String bat = Platform.isWindows ? '.bat' : '';
+
+    final ArgParser argParser = ArgParser();
+    argParser.addOption(
+      'dart',
+      help: 'The path to the dart binary.',
+      valueHelp: path.join(r'$FLUTTER_ROOT', 'bin', 'cache', 'dart-sdk', 'bin', 'dart$exe'),
+      defaultsTo: path.join(fs.currentDirectory.path, 'bin', 'cache', 'dart-sdk', 'bin', 'dart$exe'),
+    );
+    argParser.addOption(
+      'flutter',
+      help: 'The path to the flutter binary.',
+      valueHelp: path.join(r'$FLUTTER_ROOT', 'bin', 'flutter$bat'),
+      defaultsTo: path.join(fs.currentDirectory.path, 'bin', 'flutter$bat'),
+    );
+    argParser.addOption(
+      'gen-snapshot',
+      help: 'The path to gen_snapshot.',
+      valueHelp: path.join(r'$FLUTTER_ROOT', 'bin', 'cache', 'artifacts', 'engine', 'android-arm64-release', _getPlatformName(), 'gen_snapshot$exe'),
+      defaultsTo: path.join(fs.currentDirectory.path, 'bin', 'cache', 'artifacts', 'engine', 'android-arm64-release', _getPlatformName(),'gen_snapshot$exe'),
+    );
+    argParser.addOption(
+      'fe-snapshot',
+      help: 'The path to the frontend server snapshot.',
+      valueHelp: path.join(
+          r'$FLUTTER_ROOT', 'bin', 'cache', 'artifacts', 'engine', _getPlatformName(), 'frontend_server.dart.snapshot'),
+      defaultsTo: path.join(fs.currentDirectory.path, 'bin', 'cache', 'artifacts', 'engine', _getPlatformName(),
+          'frontend_server.dart.snapshot'),
+    );
+    argParser.addOption(
+      'target',
+      abbr: 't',
+      valueHelp: path.join(r'$FLUTTER_ROOT', 'examples', 'hello_world', 'lib', 'main.dart'),
+      defaultsTo: path.join(fs.currentDirectory.path, 'examples', 'hello_world', 'lib', 'main.dart'),
+    );
+    argParser.addOption(
+      'sdk-product',
+      help: 'The flutter patched sdk.',
+      valueHelp:
+          path.join(r'$FLUTTER_ROOT', 'bin', 'cache', 'artifacts', 'engine', 'common', 'flutter_patched_sdk_product'),
+      defaultsTo: path.join(
+          fs.currentDirectory.path, 'bin', 'cache', 'artifacts', 'engine', 'common', 'flutter_patched_sdk_product'),
+    );
+    argParser.addOption(
+      'package-config',
+      valueHelp: path.join(r'$FLUTTER_ROOT', 'examples', 'hello_world', '.dart_tool', 'package_config.json'),
+      defaultsTo: path.join(fs.currentDirectory.path, 'examples', 'hello_world', '.dart_tool', 'package_config.json'),
+    );
+
+    argParser.addOption(
+      'package-name',
+      valueHelp: 'package:hello_world/main.dart',
+      defaultsTo: 'package:hello_world/main.dart',
+    );
+
+    argParser.addMultiOption(
```

Comment:
This is going to have some limitations, in case we add some classes private or otherwise that have similar names. This could include a library URI to disambiguate further

Reply:
I've redone this so that it requires a package URI